src/engine/surface_collision.h

```diff
@@ -49,7 +50,11 @@ f32 find_ceil(f32 posX, f32 posY, f32 posZ, struct Surface **pceil);
 
 // Finds the ceiling from a vec3f and a minimum height (with 3 unit vertical buffer).
 ALWAYS_INLINE f32 find_mario_ceil(Vec3f pos, f32 height, struct Surface **ceil) {
-    return find_ceil(pos[0], MAX(height, pos[1]) + 3.0f, pos[2], ceil);
+#ifdef FIX_EXPOSED_CEILINGS
```

Comment:
this is an ugly way to do this, id rather you just have `height = MAX(height, pos[1]) + 3.0f;` in an ifdef

Reply:
Done, but I moved the `+ 3.0f` outside the ifdef since vanilla also has the offset.